packages/components/managed-data-table/src/data-table-settings/data-table-settings.spec.js

```diff
@@ -0,0 +1,183 @@
+        const props = createTestProps({
+          displaySettings: { disableDisplaySettings: true },
+          columnManager: { disableColumnManager: true },
+        });
+
+        const { container } = render(<DataTableSettings {...props} />);
+        expect(
+          container.querySelector("[name='table-settings-dropdown']")
+        ).not.toBeInTheDocument();
+      });
+    });
+
+    it('display settings enabled', () => {
+      const options = getDropdownOptions({
+        isColumnManagerEnabled: true,
+        isDisplaySettingsEnabled: true,
+        formatMessage,
+      });
+
+      expect(options).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            value: DISPLAY_SETTINGS,
+          }),
+        ])
+      );
+    });
+
+    it('display settings disabled', () => {
+      const options = getDropdownOptions({
+        isColumnManagerEnabled: true,
+        isDisplaySettingsEnabled: false,
+        formatMessage,
+      });
+
+      expect(options).not.toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            value: DISPLAY_SETTINGS,
+          }),
+        ])
+      );
+    });
+
+    it('column manager enabled', () => {
+      const options = getDropdownOptions({
+        isColumnManagerEnabled: true,
+        isDisplaySettingsEnabled: true,
+        formatMessage,
+      });
+
+      expect(options).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            value: COLUMN_MANAGER,
+          }),
+        ])
+      );
+    });
+
+    it('column manager disabled', () => {
+      const options = getDropdownOptions({
+        isColumnManagerEnabled: false,
+        isDisplaySettingsEnabled: true,
+        formatMessage,
+      });
+
+      expect(options).not.toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            value: COLUMN_MANAGER,
+          }),
+        ])
+      );
+    });
+  });
+
+  it('table settings topBar component', () => {
+    const topBar = 'Top Bar';
+    const props = createTestProps({ topBar });
+    render(<DataTableSettings {...props} />);
+
+    expect(screen.getByText(topBar)).toBeInTheDocument();
+  });
+
+  describe('display settings panel', () => {
+    it('should render display settings panel', () => {
+      const props = createTestProps({
+        columnManager: { disableColumnManager: true },
+      });
+
+      const { container } = render(<DataTableSettings {...props} />);
+      const dropDown = container.querySelector(
+        "[name='table-settings-dropdown']"
+      );
+
+      fireEvent.focus(dropDown);
+      fireEvent.keyDown(dropDown, { key: 'ArrowDown' });
+      fireEvent.keyUp(dropDown, { key: 'ArrowDown' });
+      fireEvent.keyDown(dropDown, { key: 'Enter' });
+      fireEvent.keyUp(dropDown, { key: 'Enter' });
+
+      expect(screen.getByTestId(DISPLAY_SETTINGS)).toBeInTheDocument();
```

Comment:
Be careful because fireEvent might trigger asynchronous operations, which requires then to `await` for certain elements to appear on screen.